src/ucs/Makefile.am

```diff
@@ -15,7 +15,7 @@ bin_PROGRAMS        =
 libucs_la_CPPFLAGS = $(BASE_CPPFLAGS) $(BFD_CPPFLAGS) \
                      -DUCX_MODULE_DIR=\"$(moduledir)\" \
                      -DUCX_CONFIG_DIR=\"$(ucx_config_dir)\"
-libucs_la_CFLAGS   = $(BASE_CFLAGS) $(BFD_CFLAGS)
+libucs_la_CFLAGS   = $(BASE_CFLAGS) $(BFD_CFLAGS) $(CFLAGS_NO_DEPRECATED)
```

Comment:
what in UCS is deprecated? maybe we can avoid it?
maybe we can put isnan+isinf inside a preprocessor macro that check is fast-path float is disabled?

Reply:
It is not isnan/isinf, it is ".. use readdir() instead of readdir_r() which is deprecated ..", coming from added `icx` builds.

Reply:
> It is not isnan/isinf, it is ".. use readdir() instead of readdir_r() which is deprecated ..", coming from added `icx` builds.

so maybe let's change to readdir_r? IMO it's better to fix than adding compiler flags to ignore

Reply:
it's the other way around, ucs uses readdir_r(). we'd need to move to readdir(), which appears to not always be reentrant for it's internal buffers, depending on libcs/versions, so I'd keep that change for another PR.

Reply:
According to https://man7.org/linux/man-pages/man3/readdir_r.3.html, seems we can use readdir() as it is also thread-safe in modern implementations.

Reply:
ok, replaced with `readdir()`.